src/js/pwm.js

```diff
@@ -97,77 +89,84 @@ function PWM(pin, options, callback) {
   });
 }
 
-// pwm.setPeriod(period[, callback])
-PWM.prototype.setPeriod = function (period, callback) {
-  var self = this;
-  // Check arguments.
+PWM.prototype._validatePeriod = function(period) {
   if (!util.isNumber(period)) {
-    throw new TypeError('Bad arguments - period');
+    throw new TypeError('Period is not a number(' + typeof(period) + ')');
+  } else if (period < 0) {
+    throw new RangeError('Period(' + period + ') is negative');
   }
+  return true;
+};
+
+PWM.prototype._validateDutyCycle = function(dutyCycle) {
+  if (!util.isNumber(dutyCycle)) {
+    throw TypeError('DutyCycle is not a number(' + typeof(dutyCycle) + ')');
+  } else if (dutyCycle < 0.0 || dutyCycle > 1.0) {
```

Comment:
Same here -- if the dutyCycle is not a number, it should not be compared against 0 and 1.